packages/opds-web-client/src/components/Root.tsx

```diff
@@ -323,28 +340,11 @@ export class Root extends React.Component<RootProps, RootState> {
                     preferences={this.props.preferences}
                     setPreference={this.props.setPreference}
                   />
-                </CollectionContainer>
-              ) : (
-                <Collection
-                  collection={this.collectionDataWithLoans()}
-                  fetchPage={this.props.fetchPage}
-                  isFetchingCollection={this.props.isFetchingCollection}
-                  isFetchingBook={this.props.isFetchingBook}
-                  isFetchingPage={this.props.isFetchingPage}
-                  error={this.props.error}
-                  updateBook={this.props.updateBook}
-                  fulfillBook={this.props.fulfillBook}
-                  indirectFulfillBook={this.props.indirectFulfillBook}
-                  isSignedIn={this.props.isSignedIn}
-                  epubReaderUrlTemplate={this.props.epubReaderUrlTemplate}
-                  preferences={this.props.preferences}
-                  setPreference={this.props.setPreference}
-                />
-              )
-            ) : null}
+                )
+              ) : null)}
           </div>
         </main>
-        {showFooter && (
+        {Footer && this.props.collectionData && (
```

Comment:
Any reason why the variable was removed in favor for this? It seems `showFooter` was the same thing.

Reply:
Yea, my comment somewhere explains it in detail. Typescript can't store a "type guard" in a variable, so `showFooter` worked, but typescript didn't understand that that variable meant both `Footer` and `this.props.collectionData` were defined. This was always the case, it's just that before `strictNullChecks`, typescript was fine with accessing properties on possibly undefined vars. 

So long story short, in order to allow typescript to refine the type so that it knows those vars are defined, we have to perform the check inline, or write a type guard function that does it. I could have turned `showFooter` into one of those functions, but decided this was simpler. 

Hopefully one day we will be able to type a boolean like this:
```js
const showFooter: Footer is JSX.Element = !!Footer;
```